lib/public/components/Table/SortModel.js

```diff
@@ -0,0 +1,158 @@
+ * or submit itself to any jurisdiction.
+ */
+import { Observable } from '/js/src/index.js';
+
+const ORDERS = [null, 'asc', 'desc'];
+
+/**
+ * Model storing sort state
+ */
+export class SortModel extends Observable {
+    /**
+     * Simple constructor
+     */
+    constructor() {
+        super();
+
+        // eslint-disable-next-line prefer-destructuring
+        this._appliedDirection = this.defaultOrder;
+
+        this._appliedOn = null;
+
+        this._previewOn = null;
+    }
+
+    /**
+     * Return the direction of the sort applied on a given column
+     *
+     * @param {string} column the column key for which sort direction is queried
+     *
+     * @return {string|null} the applied direction
+     */
+    getAppliedDirection(column) {
+        return column === this._appliedOn ? this._appliedDirection : this.defaultOrder;
+    }
+
+    /**
+     * Return the preview direction applied on a given column
+     *
+     * @param {string} column the key of the column for which preview direction must be returned
+     *
+     * @return {string|null} the preview direction
+     */
+    getPreviewDirection(column) {
+        if (!this.isPreviewOn(column)) {
+            return null;
+        }
+
+        return this._previewOn && this.getNextOrder(column === this._appliedOn ? this._appliedDirection : this.defaultOrder);
+    }
+
+    /**
+     * Define the new current direction and column by cycling the current one
+     *
+     * @param {string} column the column key on which sorting must apply
+     *
+     * @return {void}
+     */
+    cycleDirection(column) {
+        if (column !== this._appliedOn) {
+            this._appliedOn = column;
+            this._appliedDirection = this.defaultOrder;
+        }
+
+        this._appliedDirection = this.getNextOrder(this._appliedDirection);
+
+        if (this._appliedDirection === null) {
+            this._appliedOn = null;
+        }
+
+        this.notify();
+    }
+
+    /**
+     * Returns the order right after the given one (cycling)
+     *
+     * @param {string|null} order the current order
+     *
+     * @return {string|null} the next order
+     */
+    getNextOrder(order) {
+        const newDirectionIndex = (ORDERS.findIndex((direction) => direction === order) + 1) % ORDERS.length;
+        return ORDERS[newDirectionIndex];
+    }
+
+    /**
+     * States if there is a preview on the given column
+     *
+     * @param {string} column the column on which there is a preview
+     *
+     * @return {boolean} true if the column has the sort preview
+     */
+    isPreviewOn(column) {
+        return column === this._previewOn;
+    }
+
+    /**
+     * Define the column on which preview must be enabled
+     *
+     * @param {string} column the key of the column on which preview must be enabled
+     *
+     * @return {void}
+     */
+    setPreviewOn(column) {
+        this._previewOn = column;
+        this.notify();
+    }
```

Comment:
Is there a reason why some getters and setters are not following ES standards? If not can it be changed as you did on `appliedDirection`?

Reply:
Not really, I changed it